app/resources/scanned_maps/scanned_maps_controller.rb

```diff
@@ -3,4 +3,13 @@ class ScannedMapsController < ScannedResourcesController
   include GeoResourceController
   include GeoblacklightDocumentController
   self.resource_class = ScannedMap
+
+  # View the structural metadata for a given repository resource
+  def structure
+    @change_set = change_set_class.new(find_resource(params[:id])).prepopulate!
+    @logical_order = (Array(@change_set.logical_structure).first || Structure.new).decorate
+    members = Wayfinder.for(@change_set.resource).logical_structure_members
+    @logical_order = WithProxyForObject.new(@logical_order, members)
+    authorize! :structure, @change_set.resource
```

Comment:
Should the `authorize!` call come as early as possible (presumably right after the @change_set is instantiated)?

Reply:
Makes sense. This method was copied from the scanned resource controller and modified slightly. I can update that one as well.